staging/src/k8s.io/client-go/applyconfigurations/meta/v1/unstructured.go

```diff
@@ -0,0 +1,130 @@
+package v1
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/managedfields"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager"
+	"k8s.io/client-go/discovery"
+	"k8s.io/kube-openapi/pkg/util/proto"
+	"sigs.k8s.io/structured-merge-diff/v4/typed"
+)
+
+// UnstructuredExtractor enables extracting the applied configuration state from object for fieldManager into an
+// unstructured object type.
+type UnstructuredExtractor interface {
+	ExtractUnstructured(object *unstructured.Unstructured, fieldManager string) (*unstructured.Unstructured, error)
+	ExtractUnstructuredStatus(object *unstructured.Unstructured, fieldManager string) (*unstructured.Unstructured, error)
+}
+
+// gvkParserCache caches the GVKParser in order to prevent from having to repeatedly
+// parse the models from the open API schema when the schema itself changes infrequently.
+type gvkParserCache struct {
+	// discoveryClient is the client for retrieving the openAPI document and checking
+	// whether the document has changed recently
+	discoveryClient discovery.DiscoveryInterface
+	// ttl is how long the openAPI schema should be considered valid
+	ttl time.Duration
```

Comment:
OK Got it now, you made the time configurable. I don't know if that's super important and I would consider making this a constant. Otherwise, I find the comment above a little confusing.

Reply:
Agree w/ Antoine unless this is used for testing.

Reply:
done